### PR TITLE
Add CapacityError::new

### LINKED ISSUE #69: Expose CapacityError::new
It's already possible for a user to create this on their own:

```
fn cap_err<T>(val: T) -> CapacityError<T> {
    let mut av: ArrayVec<[T; 0]> = ArrayVec::new();
    av.try_push(val).unwrap_err()
}
```

I currently use a wrapper over `ArrayVec` for my own type and it'd be nice to create my own `CapacityError`s without having to create my own type.

### PR DESCRIPTION
**Pull request by bluss**
Fixes #69

### CODE REVIEW

**Comment by bluss** [comment]
cc @tbu-

File: src/errors.rs [hunk 1]
```diff
@@ -10,20 +10,15 @@ pub struct CapacityError<T = ()> {
...
+    pub fn new(element: T) -> CapacityError<T> {
         CapacityError {
             element: element,
         }
     }
-}
-
-impl<T> CapacityError<T> {
     /// Extract the overflowing element
+    ///
```
Comment (tbu-) [inline src/errors.rs]: Extra line.

**Comment by tbu-** [comment]
LGTM

File: src/errors.rs (hunk 1 above)
Comment (bluss) [reply src/errors.rs]: Thanks

**Comment by bluss** [comment]
Merged with the extra line removed

**Pull request merged**